generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management.tsx.ejs

```diff
@@ -39,27 +39,17 @@ export interface IUserManagementProps extends StateProps, DispatchProps, RouteCo
 export const UserManagement = (props: IUserManagementProps) => {
   const [pagination, setPagination] = useState(getSortState(props.location, ITEMS_PER_PAGE));
 
-  const getAllUsers = () =>
-    props.getUsers(pagination.activePage - 1, pagination.itemsPerPage, `${pagination.sort},${pagination.order}`);
-
-  useEffect(() => {
-    getAllUsers();
-  }, []);
-
-  const sortUsers = () => getAllUsers();
-
   useEffect(() => {
-    sortUsers();
-  }, [pagination.activePage, pagination.order, pagination.sort]);
+    props.getUsers(pagination.activePage - 1, pagination.itemsPerPage, `${pagination.sort},${pagination.order}`)
+    props.history.replace(`${props.location.pathname}?page=${pagination.activePage}&sort=${pagination.sort},${pagination.order}`);
```

Comment:
any specific reason to `replace` history?

Reply:
Thanks for spoting this! 
Indeed, to be coherent with Angular version it should be a **history.push**.
I will change this.

However with a history.push, when we _go back_ with the browser, the previous state sort is not applied (but the url is updated).
By using **history.replace** we will go back to the previous page no matter all the sorts that we could have performed on the page.

So I will try to see how we can fix this (in Angular, React and maybe Vue.js) in another PR.